src/hooks/useAutoSlippageTolerance.ts

```diff
@@ -14,14 +14,28 @@ import useStablecoinPrice, { useStablecoinValue } from './useStablecoinPrice'
 const V3_SWAP_DEFAULT_SLIPPAGE = new Percent(50, 10_000) // .50%
 const ONE_TENTHS_PERCENT = new Percent(10, 10_000) // .10%
 export const DEFAULT_AUTO_SLIPPAGE = ONE_TENTHS_PERCENT
+const GAS_ESTIMATE_BUFFER = new Percent(10, 100) // 10%
 
 /**
  * Return a guess of the gas cost used in computing slippage tolerance for a given trade
  * @param trade the trade for which to _guess_ the amount of gas it would cost to execute
  */
 function guesstimateGas(trade: Trade<Currency, Currency, TradeType> | undefined): number | undefined {
   if (!!trade) {
-    return 100_000 + trade.swaps.reduce((memo, swap) => swap.route.pools.length + memo, 0) * 30_000
+    let gas = 0
+    for (const { route } of trade.swaps) {
+      if (route.protocol === Protocol.V2) {
+        gas += 135000 + route.pools.length * 50_000
```

Comment:
can you name variables for `135000` and `50_000`?

Reply:
i think you can actually do something like this...
```ts
let baseGas = 0
let perRouteGas = 0

if (route.protocol === Protocol.V2) {
  baseGas = 135000
  perRouteGas = 50_000
}

gas += baseGas + route.pools.length * perRouteGas
```

something like that

Reply:
Each route could use a different protocol so I didn't go with what you suggested. I moved the values into constants though

Reply:
works for me